cdap-ui/app/cdap/components/Header/Welcome/index.tsx

```diff
@@ -0,0 +1,158 @@
+// to show the guided tour again.
+const USER_STORE_KEY = 'showWelcome';
+const USER_STORE_VALUE = 1;
+const SESSION_STORAGE_KEY = USER_STORE_KEY;
+const SESSION_STORAGE_VALUE = 'true';
+const PREFIX = 'features.NUX.Welcome';
+
+export default class Welcome extends React.PureComponent {
+  public state = {
+    showModal: false,
+    showAgain: false,
+  };
+
+  public componentDidMount() {
+    // Checking for session storage so that the welcome modal is not shown
+    // when user is navigating back and forth between react and angular pages.
+    const sessionValue = window.sessionStorage.getItem(SESSION_STORAGE_KEY);
+    if (sessionValue === SESSION_STORAGE_VALUE) { return; }
+
+    MyUserStoreApi.get()
+      .subscribe((res) => {
+        const storeValue = objectQuery(res, 'property', USER_STORE_KEY);
+
+        if (!storeValue || storeValue !== USER_STORE_VALUE) {
+          this.setState({
+            showModal: true,
+          });
+        }
+      });
+  }
+
+  private close = () => {
+    this.setState({
+      showModal: false,
+    });
+
+    window.sessionStorage.setItem(SESSION_STORAGE_KEY, SESSION_STORAGE_VALUE);
+
+    this.saveUserState();
+  }
+
+  private startTour = () => {
+    this.close();
+    Tour.start();
+  }
+
+  private toggleShowAgain = () => {
+    this.setState({
+      showAgain: !this.state.showAgain,
+    });
+  }
+
+  private saveUserState = () => {
+    if (!this.state.showAgain) { return; }
+
+    MyUserStoreApi.get()
+      .subscribe((res) => {
+        const obj = {
+          ...res.property,
+        };
+
+        obj[USER_STORE_KEY] = USER_STORE_VALUE;
+
+        MyUserStoreApi.set(null, obj);
+      });
+  }
+
+  public render() {
+    return (
+      <If condition={this.state.showModal}>
+        <Modal
+          isOpen={true}
+          size="md"
+          backdrop={false}
+          centered={true}
+          zIndex="1061"
+          className="welcome-modal"
+        >
+          <ModalHeader>
+            <span className="header-text">
+              {T.translate(`${PREFIX}.header`)}
+            </span>
+
+            <div className="close-section float-xs-right" >
+              <IconSVG
+                name="icon-close"
+                onClick={this.close}
+              />
+            </div>
+          </ModalHeader>
+
+          <ModalBody>
+            <p>
+              {T.translate(`${PREFIX}.bodyText`)}
+            </p>
+            <p>
+              {T.translate(`${PREFIX}.takeTour`)}
+            </p>
+
+            <div className="show-again-selection">
+              <span onClick={this.toggleShowAgain}>
+                <IconSVG name={this.state.showAgain ? 'icon-check-square' : 'icon-square-o'} />
+
+                <span>
+                  {T.translate(`${PREFIX}.showAgainToggle`)}
+                </span>
+              </span>
+            </div>
+
+            <div className="action-buttons">
```

Comment:
We right now follow a pattern of having a footer with primary and secondary buttons. We should follow the same pattern.